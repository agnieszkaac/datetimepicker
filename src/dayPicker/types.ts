import React from "react";

export interface DayProps {
  value: string;
  selected: boolean;
  onPick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export interface DayPickerProps {
  value: string;
  monthLength: number;
  onPick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}
