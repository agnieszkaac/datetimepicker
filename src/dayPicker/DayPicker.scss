$size: 25px;

.day-picker {
  display: grid;
  grid-template-columns: repeat(7, $size);
  grid-gap: 3px;
}

.day {
  display: flex;
  justify-content: center;
  align-items: center;
  height: $size;
  width: $size;
  background-color: white;
  color: black;
  border: none;
  outline: none;

  &:hover {
    background-color: silver;
  }
  &.selected {
    background-color: aquamarine;
  }
  &.is-today {
    border: 1px solid blue;
  }
}

