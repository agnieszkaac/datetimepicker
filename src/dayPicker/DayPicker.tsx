import React from "react";
import moment from "moment";

import { DayPickerProps, Day } from "./";

export const DayPicker: React.FunctionComponent<DayPickerProps> = ({
  date,
  viewDate,
  onPick,
}) => {
  return (
    <div className="day-picker">
      {Array.from({ length: moment(viewDate).daysInMonth() }, (_e, i) => (
        <Day
          key={i + 1}
          date={moment(viewDate).date(i + 1)}
          selected={date?.date() === i + 1 && date?.isSame(viewDate, "month")}
          onPick={onPick}
        />
      ))}
    </div>
  );
};
