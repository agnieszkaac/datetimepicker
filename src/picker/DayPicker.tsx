import React from "react";
import cx from "classnames";
import moment from "moment";

import { DayPickerProps, DayProps } from "../types";

const Day: React.FunctionComponent<DayProps> = ({
  value,
  selected,
  isToday,
  onPick,
}) => (
  <button
    className={cx("day", { selected: selected, "is-today": isToday })}
    onClick={onPick}
    value={value}
  >
    {value}
  </button>
);

export const DayPicker: React.FunctionComponent<DayPickerProps> = ({
  value,
  monthLength,
  onPick,
}) => (
  <div className="day-picker">
    {Array.from({ length: monthLength }, (e, i) => (
      <Day
        key={i + 1}
        value={(i + 1).toString()}
        selected={value === (i + 1).toString()}
        isToday={moment().date().toString() === value}
        onPick={onPick}
      />
    ))}
  </div>
);
