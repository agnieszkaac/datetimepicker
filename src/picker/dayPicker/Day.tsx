import React from "react";
import { Moment } from "moment";
import cx from "classnames";

export type DayProps = {
  viewDate: Moment;
  className?: string;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export const Day: React.FunctionComponent<DayProps> = ({
  viewDate,
  className,
  onClick,
}) => (
  <button
    type="button"
    data-testid="day"
    value={viewDate.toISOString()}
    className={cx("day", className)}
    onClick={onClick}
  >
    {viewDate.date()}
  </button>
);
