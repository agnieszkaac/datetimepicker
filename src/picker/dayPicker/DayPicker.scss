@import "../../common";

$size: $size-m;

.day {
  height: $size;
  width: $size;
  color: $text;
  background-color: $background;
  border: none;
  outline: none;
  border-radius: 2px;
  cursor: pointer;

  &:hover {
    background-color: $background-hover;
  }
}

.day-picker {
  @include grid($size, 7);
  & .day {
    &.prev-month,
    &.next-month {
      color: $text-light;
    }
    &.is-today {
      border: 1px solid $border-today;
    }
    &.selected {
      color: $text-selected;
      background-color: $background-selected;
    }
  }
}
.week-days {
  @include grid($size, 7);
  .day-of-week {
    margin: $spacing-m 0;
    text-align: center;
    font-size: $font-size-small;
    line-height: $size-xs;
    font-weight: 600;
    vertical-align: middle;
  }
}
