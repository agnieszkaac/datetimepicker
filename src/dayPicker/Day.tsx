import React from "react";
import moment from "moment";
import cx from "classnames";

import { DayProps } from "./";
import "./DayPicker.scss";

export const Day: React.FunctionComponent<DayProps> = ({
  date,
  selected,
  onPick,
}) => {
  const isToday = date.isSame(moment(), "day");
  return (
    <button
      type="button"
      className={cx("day", { selected: selected, "is-today": isToday })}
      value={date.toISOString()}
      onClick={onPick}
    >
      {date.date()}
    </button>
  );
};
