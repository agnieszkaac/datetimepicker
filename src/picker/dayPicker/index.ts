import { Day, DayProps } from "./Day";
import { DayPicker, DayPickerProps } from "./DayPicker";
import { WeekDays, WeekDaysProps } from "./WeekDays";

export { Day, DayProps, DayPicker, DayPickerProps, WeekDays, WeekDaysProps };
