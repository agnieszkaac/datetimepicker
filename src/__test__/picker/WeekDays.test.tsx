import React from "react";
import MockDate from "mockdate";
import moment, { Moment } from "moment";
import { render } from "@testing-library/react";

import { WeekDays, WeekDaysProps } from "../../picker/dayPicker";

describe("<WeekDays /> should", () => {
  let viewDate: Moment;
  let props: WeekDaysProps;

  beforeEach(() => {
    MockDate.set("2020-01-01");

    viewDate = moment();
    props = {
      viewDate,
    };
  });

  afterEach(() => {
    MockDate.reset();
  });

  it("match snapshot", () => {
    const { asFragment } = render(<WeekDays {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("render 7 week day names", () => {
    const { getAllByTestId } = render(<WeekDays {...props} />);
    const days = getAllByTestId("week-day");

    expect(days.length).toEqual(7);
  });

  it("change week day language when changed locale", () => {
    const { getByText, rerender } = render(<WeekDays {...props} />);
    const mon = getByText("Mon");

    expect(mon).toBeInTheDocument();

    viewDate.locale("pl");
    rerender(<WeekDays {...props} />);
    const pon = getByText("pon");

    expect(pon).toBeInTheDocument();
  });
});
