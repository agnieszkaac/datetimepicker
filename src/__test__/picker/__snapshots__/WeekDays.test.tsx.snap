// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeekDays /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="week-days"
  >
    <span
      class="week-day"
      data-testid="week-day"
    >
      Sun
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Mon
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Tue
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Wed
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Thu
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Fri
    </span>
    <span
      class="week-day"
      data-testid="week-day"
    >
      Sat
    </span>
  </div>
</DocumentFragment>
`;
